packages/ses/src/get-anonymous-intrinsics.js

```diff
@@ -161,5 +161,19 @@ export const getAnonymousIntrinsics = () => {
     );
   }
 
+  if (globalThis.ModuleSource) {
+    const AbstractModuleSourcePrototype = getPrototypeOf(
+      globalThis.ModuleSource.prototype,
+    );
+    intrinsics['%AbstractModuleSourcePrototype%'] =
+      AbstractModuleSourcePrototype;
+    intrinsics['%AbstractModuleSource%'] =
+      AbstractModuleSourcePrototype.constructor;
+  }
```

Comment:
A valid but annoying feedback reviewers might provide here is that we can’t anticipate whether `ModuleSource` will land in the language with or without an `AbstractModuleSource` on its prototype chain, so we could hedge our bets and add a repair for ModuleSource to force it to appear one way or the other, so that `lockdown()` doesn’t break if the AbstractModuleSource is absent. Or, we could go the other way and _not_ have AbstractModuleSource by default, in which case SES would just delete it and issue a warning if it showed up.

Reply:
You read my mind, but that's just the risk with front-running proposals. No suggestion.

Reply:
Capturing here my reasoning for keeping AbstractModuleSource since it appears to be the path of least regret.

1. We shim AbstractModuleSource and include it in SES permits:
    1. The application includes module-source/shim.js:
        1. The VM implements AbstractModuleSource: :white_check_mark: Lockdown’s fine
        2. The VM does not implement AbstractModuleSource: :white_check_mark: Lockdown’s fine. The shim currently stomps the native implementation. If it didn’t stomp the native implementation, Lockdown would tolerate the incomplete intrinsics.
    2. The application excludes module-source/shim.js:
        1. The VM implements AbstractModuleSource: :white_check_mark: Lockdown’s fine
        2. The VM does not implement AbstractModuleSource: :white_check_mark: Lockdown’s fine, but intrinsics are incomplete
2. We do not shim AbstractModuleSource and exclude it in SES permits:
    1. The application includes module-source/shim.js:
        1. The VM implements AbstractModuleSource: :white_check_mark: :man-shrugging: The shim currently stomps the native implementation, so this works out. But, if the shim surfaced the native implementation (and perhaps feature-checks that the native implementation behaves just like the shim deigning to fall through to native)
        2. The VM does not implement AbstractModuleSource: :white_check_mark: Lockdown’s fine
    2. The application excludes module-source/shim.js:
        1. The VM implements AbstractModuleSource: :broken_heart: Lockdown is surprised to find AbstractModuleSource and chooses throw rather than attempt a repair. We don’t have the module source shim in play, so there’s nothing it can do to mitigate. We can’t require the module-source/shim.js because it entrains Babel.
        2. The VM does not implement AbstractModuleSource: :white_check_mark: Lockdown’s fine